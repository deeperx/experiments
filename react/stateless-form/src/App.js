import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import './App.css';

import ContactForm from './contact-form';

class App extends Component {
  formOnChange(details) {
    console.log('details: ', details);
  }

  render() {
    return (
      <MuiThemeProvider>
        <ContactForm onSubmit={ this.formOnChange.bind(this) }/>
      </MuiThemeProvider>
    );
  }
}

export default App;
