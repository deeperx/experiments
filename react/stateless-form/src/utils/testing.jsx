import renderer from 'react-test-renderer'

export const expectToMatchSnapshot = el =>
  expect(
    renderer.create(el).toJSON()
  ).toMatchSnapshot()
