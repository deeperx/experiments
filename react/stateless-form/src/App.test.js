import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { expectToMatchSnapshot } from './utils/testing'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
})

it('renders correctly', expectToMatchSnapshot(
  <App />
))
