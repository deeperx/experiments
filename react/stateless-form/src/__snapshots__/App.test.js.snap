exports[`undefined 1`] = `
<form
  onChange={[Function]}
  onSubmit={[Function]}>
  <p>
    <label
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your name
      :
    </label>
    <input
      defaultValue=""
      name="name"
      type="text" />
  </p>
  <p>
    <label
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your lastname
      :
    </label>
    <input
      defaultValue=""
      name="lastname"
      type="text" />
  </p>
  <p>
    <label
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your email
      :
    </label>
    <input
      defaultValue=""
      name="email"
      type="text" />
  </p>
  <input
    type="submit"
    value="Submit" />
  <input
    type="reset"
    value="Clear" />
</form>
`;
